lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/EvidenceFactory.java

```diff
@@ -72,11 +72,25 @@ private CheckDetail[] createCheckDetailsWithKbvQuality(KBVItem kbvItem)
         var questionState = objectMapper.readValue(kbvItem.getQuestionState(), QuestionState.class);
 
         if (hasPassedWithOneIncorrectAnswer(kbvItem)) {
+            /**
+             * Scenario: We create CheckDetails with the lowest 3 kbvQuality values So we are
+             * excluding the highest kbvQuality value
+             *
+             * <p>1 question was wrong in the first batch And therefore 2 questions were given in
+             * second batch And they were answered correctly So we can sort by quality and remove
+             * any additional question(s)
+             */
             if (questionState.allQuestionBatchSizesMatch(2)) {
                 return createCheckDetailsBySortingOnKbvQuality(kbvItem, questionState);
             }
+            /**
+             * Scenario: 2 questions in first batch were correct 3rd question in the second batch
+             * was incorrect And 4th question in the third batch was correct So we can remove the
+             * third question because we are sure it was wrong
+             */
             return createCheckDetailsBySkipping3rdIncorrectQa(questionState);
         }
+        /** Scernaio: 3 out of 3 correct answers from 2 batches */
```

Comment:
Yes: Scenario where 2 answer correctly and therefore 1 more needed and answerd correctly